game-core/src/test/java/games/strategy/triplea/settings/GameSettingTest.java

```diff
@@ -0,0 +1,32 @@
+package games.strategy.triplea.settings;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.NoSuchElementException;
+
+import org.junit.jupiter.api.Test;
+
+class GameSettingTest {
+
+  @Test
+  @SuppressWarnings("unchecked")
```

Comment:
You can avoid having to suppress the unchecked warning by injecting the mock into the test method:

```diff
diff --git a/game-core/src/test/java/games/strategy/triplea/settings/GameSettingTest.java b/game-core/src/test/java/games/strategy/triplea/settings/GameSettingTest.java
index 9f37604d3..e14813e2a 100644
--- a/game-core/src/test/java/games/strategy/triplea/settings/GameSettingTest.java
+++ b/game-core/src/test/java/games/strategy/triplea/settings/GameSettingTest.java
@@ -3,7 +3,6 @@ package games.strategy.triplea.settings;
 import static org.junit.jupiter.api.Assertions.assertArrayEquals;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
@@ -11,13 +10,14 @@ import static org.mockito.Mockito.when;
 import java.util.NoSuchElementException;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
 class GameSettingTest {
-
+  @ExtendWith(MockitoExtension.class)
   @Test
-  @SuppressWarnings("unchecked")
-  void testGetFromSetting() {
-    final GameSetting<char[]> setting = mock(GameSetting.class);
+  void testGetFromSetting(@Mock final GameSetting<char[]> setting) {
     final char[] testData = new char[]{'a', 'b', 'c'};
     when(setting.getValueOrThrow()).thenThrow(NoSuchElementException.class).thenReturn(testData);
 
```